workflows/prognostic_c48_run/runtime/steppers/machine_learning.py

```diff
@@ -183,7 +184,17 @@ def __call__(self, time, state):
         diagnostics: Diagnostics = {}
         delp = state[DELP]
 
-        tendency: State = predict(self.model, state)
+        prediction: State = predict(self.model, state)
+
+        tendency, state_updates = {}, {}
+        for key, value in prediction.items():
+            if is_state_update_variable(key, state):
+                state_updates[key] = value
+            else:
+                tendency[key] = value
+
+        for name in state_updates.keys():
+            diagnostics[name] = state_updates[name]
 
         dQ1_initial = tendency.get("dQ1", xr.zeros_like(state[SPHUM]))
         dQ2_initial = tendency.get("dQ2", xr.zeros_like(state[SPHUM]))
```

Comment:
The precipitative model is going to have some potentially confusing interactions with the non-negative specific humidity limiter, though this is not really any different from the biases the limiter introduces in other models and there's nothing I can think to easily be done about it.